### ISSUE TITLE
Player Housing 2

### ISSUE THREAD

**Issue by Valignus** [suriyun-production/mmorpg-kit-docs#1987]
Player Housing Version 2:

Instanced player housing where players can build and customize their house and surrounding land then be able to invite other players to their instanced house.
Being able to plant crops and raise animals could be useful.

There could also be an option for an instanced guild house that is customizable.

Instances could have an owner of either a player or a guild.

**Comment by juniormagalhaes** [comment]
guild house would be nice

**Comment by moepi2k** [comment]
Instanced housing would be nice. Maybe  create a poll for this as a paid request :)

**Comment by insthync** [comment]
Before I do this, I want to improve the map spawn server, and map server spawning workflow, I want to make it work similarly to Agornes, which game-servers can having to allocate, so it can run empty game-servers to prepare for player connection then when player connect, players won't have to wait for spawning.

**Comment by moepi2k** [comment]
sounds really good. another thing regarding building system are the building objects. i noticed that mapserver becomes really fast when you have alot of objects. and alot not means like flooding the server with 10k of objects. it already begins to lag when you have around 500-1000 objects.

atm sometimes i play sons of the forest, and i was wondering how the entitie worklflow in that game is handled. since every singel tree, bush, stone etc is an entitie. it seems the whole map has around 1 million of interactable entities.

once i followed the the building object in kit, and somewhere when i remember correctly was something i found that a static object has a update function which is called even when noone has interacting with it.

just wondering if that can be improved when expending the housing thing? maybe not so important for instanced housing but it is for open world building system

**Comment by Fnatic91** [comment]
Hello!

Any news about this function?

Thanks!

**Comment by LordPhrozen** [comment]
This is a *HIGHLY* desired feature, for sure, in any implementation. Eagerly awaiting this add as well.

**Comment by insthync** [comment]
If it charges 2,000 USD, pay once and it will be released for everyone, will you guys pay for it?

**Comment by moepi2k** [comment]
i would do it that way. last time u did a poll in ko-fi for auction house. make same for feature request so we can share the link in discord so community can spend money to it.

**Comment by Fnatic91** [comment]
Well $2k USD sounds a bit too much just for this one feature especially if we takes that the entire kit cost like $80, however i'm willing to pay like $200, incase if its being worked on and being added asap and both Player Housing 1 & 2 are being added! Hopefully others are can/willing to do similar!

**Comment by LordPhrozen** [comment]
@insthync 

I kind of agree with @Fnatic91 , $2k for a feature that's then available to the public after, and will drive additional sales, seems a bit steep, especially since the instancing tech is already in place.

Now that being said, I also want to see more money in your pocket, this is a fantastic product, and it deserves to generate you more income. Neomis had a fantastic idea in the discord, and that's for **additional major features to be separate addon's purchased from the Unity Asset Store.**

A $30-50 price tag on a major feature and selling it there, seems completely reasonable, and will continue to generate revenue long term. How do you feel about doing it that way?


Additionally, if what @moepi2k is mentioning here is true, that theres performance issues with building parts, that would likely need addressing first.

**Comment by Fnatic91** [comment]
I agree, the best would be that new features/packages could be sold in the asset store separatedly for different prices, but my point is that of course we dosn't wants @insthync to work continously for free at all and as i mentioend i even willing to pay like $200 just to have this feature fast.

**Comment by Kanera66** [comment]
hello, I never comment on github or on the MMORPG kit but I come here often (since 1 year) to see if this feature will be added. 

I'm also willing to pay to see this functionality added. this would solve a lot of optimization issues and would greatly affect gameplay.
(I saw the video on your youtube channel with the Channels) we are getting closer to what an instanced housing could be. thank you for all

**Comment by moepi2k** [comment]
as @LordPhrozen already mention above, we discused it a little bit in discord. i dont know whats your plan, and its up to you. in any case, it should be worthwhile for you to make new features. the idea of selling bigger features as sepearate addons in the asset store would make you money in the future and also not overload the kit. since not everyone needs every feature.

**Comment by darkkriteus** [comment]
this is completely individual from game to game. my opinion and focus on server improvements, and fixing bugs, this could be a separate plugin or addon just for those who want to use it

**Comment by insthync** [comment]
Some of you told me to focus on server improvements and bug fixing but still requesting new features...

**Comment by darkkriteus** [comment]
> Some of you told me to focus on server improvements and bug fixing but still requesting new features...

Yes, that's true, I recently requested some improvements, possible and small improvements I think are valid, but more complex systems that are more relative from game to game, are not the focus, as people mentioned a possible addon sold separately from the kit it would be better for the kit and for you to earn more

**Comment by LordPhrozen** [comment]
> Some of you told me to focus on server improvements and bug fixing but still requesting new features...

**I for the most part agree. Bug Fixes > Performance Improvements > New Features.**

I think the sentiment here is that this feature is just... very widely desired, and likely a simple implementation for you. 


_Here's a simple design for it. A great start for others to implement game specific ideas._

[DATA SAVED]
-Store a map/scene for this players home.
-Store a list of building objects and their rotations on the player for their "home" when they make changes inside it.
-Store an integer on the mapinfo for max placed objects.

[FUNCTIONALITY]
-Load the home's owner's objects when you load in.
-Consumable item to set your home scene. ( Confirmation on use, that any previous house building objects will be lost. )
-A way for players to enter whatever home scene they have set via the item. ( NPC and Warp )
-Party members when trying to move to their home, move to the party leaders home instead ( and load their objects).
-Dont allow placement above max objects for that map, if the maxObjects > 0 or if the home isnt yours.

**Comment by jprules321** [comment]
I was looking for exactly this. I'll chip in $300 USD if we can have this year.
